src/js/components/TooltipComponent.jsx

```diff
@@ -0,0 +1,71 @@
+import Tooltip from 'material-ui/internal/Tooltip'
+import React, {Component, PropTypes} from 'react'
+
+export default class TooltipComponent extends Component {
+  static propTypes = {
+    children: PropTypes.any
+  }
+  
+  static defaultProps = {
+    componentProps: {}
+  }
+
+  constructor (props) {
+    super(props)
+    
+    this.state = {showTooltip: false}
+
+    this.handleBlur = ::this.handleBlur
+    this.handleFocus = ::this.handleFocus
+    this.handleMouseLeave = ::this.handleMouseLeave
+    this.handleMouseEnter = ::this.handleMouseEnter
+  }
+  
+  hideTooltip () {
+    this.setState({showTooltip: false})
+  }
+  
+  showTooltip () {
+    this.setState({showTooltip: true})
+  }
+  
+  handleBlur (event) {
+    this.hideTooltip()
+  }
+
+  handleFocus (event) {
+    this.showTooltip()
+  }
+  
+  handleMouseLeave (event) {
+    this.hideTooltip()
+  }
+
+  handleMouseEnter (event) {
+    this.showTooltip()
+  }
+
+  render () {
+    const {children} = this.props
+    const {showTooltip} = this.state
+
+    return (
+      <span
+        style={{position: 'relative', display: 'inline-block'}}
+        onBlur={this.handleBlur}
+        onFocus={this.handleFocus}
+        onMouseEnter={this.handleMouseEnter}
+        onMouseLeave={this.handleMouseLeave}
+      >
+        {children}
+        <Tooltip
+          horizontalPosition='center'
+          show={showTooltip}
+          touch={true}
+          verticalPosition='bottom'
+          {...this.props}
```

Comment:
This is strange to me. We take in a prop called `componentProps` but we only spread out the props object. This means that we may get unwanted props getting passed. We should probably just take in similar props as `Tooltip` from material-ui and just pass those on individually.